diff -ruN --exclude=.cargo-ok a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml	2026-08-17 17:12:45
+++ b/Cargo.toml	2026-08-17 17:12:45
@@ -41,6 +41,8 @@
 
 [dependencies.object_store]
 version = "0.14.0"
+default-features = false
+features = ["tokio"]
 
 [dependencies.rand]
 version = "0.9.2"
@@ -56,14 +58,6 @@
 [dependencies.thread_local]
 version = "1.1.9"
 
-[dependencies.tokio]
-version = "1.47.1"
-features = [
-    "macros",
-    "rt",
-    "time",
-]
-
 [dev-dependencies.tokio]
 version = "1.47.1"
 features = [
@@ -72,3 +66,16 @@
     "time",
     "test-util",
 ]
+
+[target.'cfg(target_arch = "wasm32")'.dependencies.tokio]
+package = "tokio_with_wasm"
+version = "0.10.1"
+features = ["macros", "rt", "time"]
+
+[target.'cfg(target_arch = "wasm32")'.dependencies.send_wrapper]
+version = "0.6.0"
+features = ["futures"]
+
+[target.'cfg(not(target_arch = "wasm32"))'.dependencies.tokio]
+version = "1.47.1"
+features = ["macros", "rt", "time"]
diff -ruN --exclude=.cargo-ok a/src/clock.rs b/src/clock.rs
--- a/src/clock.rs	2026-08-17 17:12:45
+++ b/src/clock.rs	2026-08-17 17:12:45
@@ -163,7 +163,14 @@
     }
 
     fn sleep<'a>(&'a self, duration: Duration) -> Pin<Box<dyn Future<Output = ()> + Send + 'a>> {
-        Box::pin(tokio::time::sleep(duration))
+        #[cfg(not(target_arch = "wasm32"))]
+        {
+            Box::pin(tokio::time::sleep(duration))
+        }
+        #[cfg(target_arch = "wasm32")]
+        {
+            Box::pin(send_wrapper::SendWrapper::new(tokio::time::sleep(duration)))
+        }
     }
 
     fn ticker<'a>(&'a self, duration: Duration) -> SystemClockTicker<'a> {
